p2p/starknet/bytereader.go

```diff
@@ -0,0 +1,21 @@
+package starknet
+
+import (
+	"io"
+
+	"google.golang.org/protobuf/encoding/protodelim"
+)
+
+var _ protodelim.Reader = (*byteReader)(nil)
+
+type byteReader struct {
+	io.Reader
+}
+
+func (r *byteReader) ReadByte() (byte, error) {
+	var b [1]byte
+	if _, err := r.Read(b[:]); err != nil {
```

Comment:
> I assume this will block until a byte is received, rather than waiting for a complete message. Is that correct?

That is my assumption as well, but I dont think it matters. We are actually interested in the message anyways, we dont really need to know what the first byte is before the entire message arrives.

> Why is this "streaming" method better than the previous non-streaming method?

There is a soft limit of 1MB for protobuf messages. 

``` proto
message List {
 repeated Elems
}
```

When we use `repeated` it essentially allows arbitrarily large messages to be created. Instead of `repeated`, we stream individual `Elems` one by one. Assuming that Elems size is constant, we for sure know that each message is below 1MB in size.

> Is there a name for the "non-streaming" method?

I dont think so